pkg/controller/sdk/provider.go

```diff
@@ -30,6 +31,7 @@ import (
 type KnativeReconciler interface {
 	Reconcile(ctx context.Context, object runtime.Object) (runtime.Object, error)
 	InjectClient(c client.Client) error
+	//TODO(grantr) make these separate interfaces
 	InjectConfig(c *rest.Config) error
```

Comment:
Does anything need this anymore? I thought it was only used for the dynamic client.

Reply:
I had originally pulled this out into a separate PR, but it was complicated to make this change separately. Further commits forthcoming.

Reply:
/hold cancel

PTAL.